[Lens] Display legend inside chart

stratoula opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/95887

This PR adds more options to the legend popover of the XY axis charts to enable the users to display the legend inside the chart.

Options that are added:
- Location button group. Users can select if they want their legend to appear inside or outside (existing functionality) the chart
- In the case of inside there are additional settings:
    - Configure the vertical and horizontal alignments
    - Configure the number of floating columns

All settings are disabled if the `hide` display option is selected.

![image](https://user-images.githubusercontent.com/17003240/126158487-f2c1c3f3-e884-4827-9e04-325a6610ade5.png)


![image](https://user-images.githubusercontent.com/17003240/125742970-1a2592f1-9f00-4a6b-abb2-53e548f890ac.png)
![image](https://user-images.githubusercontent.com/17003240/125744253-1122964d-8f7b-4452-b5a7-cbea1d9ccf7c.png)


### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [x] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/)
* Commit: fd994e6a382edc722001e3657ed7593c703986d2
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105571/fd994e6a382edc722001e3657ed7593c703986d2)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/execution/node/190/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/","number":"138408","commit":"fd994e6a382edc722001e3657ed7593c703986d2"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/)
* Commit: 6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105571/6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/ff0b0e984cc8f599646289a55d66ca0b89be355a) | [after](https://github.com/elastic/kibana/commit/6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273) | diff |
| --- | --- | --- | --- |
| `lens` | 703 | 704 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ff0b0e984cc8f599646289a55d66ca0b89be355a) | [after](https://github.com/elastic/kibana/commit/6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273) | diff |
| --- | --- | --- | --- |
| `lens` | 1.5MB | 1.5MB | +9.1KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/ff0b0e984cc8f599646289a55d66ca0b89be355a) | [after](https://github.com/elastic/kibana/commit/6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273) | diff |
| --- | --- | --- | --- |
| `lens` | 181 | 185 | +4 |


</details>

### History
* :broken_heart: [Build #138408](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/) failed fd994e6a382edc722001e3657ed7593c703986d2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/","number":"138408","commit":"fd994e6a382edc722001e3657ed7593c703986d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/","number":"138430","commit":"6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/)
* Commit: e01f85d0de69f6a5f05dde1a51b3312e76a25cca
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105571/e01f85d0de69f6a5f05dde1a51b3312e76a25cca)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/ff0b0e984cc8f599646289a55d66ca0b89be355a) | [after](https://github.com/elastic/kibana/commit/e01f85d0de69f6a5f05dde1a51b3312e76a25cca) | diff |
| --- | --- | --- | --- |
| `lens` | 703 | 704 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ff0b0e984cc8f599646289a55d66ca0b89be355a) | [after](https://github.com/elastic/kibana/commit/e01f85d0de69f6a5f05dde1a51b3312e76a25cca) | diff |
| --- | --- | --- | --- |
| `lens` | 1.5MB | 1.5MB | +9.3KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/ff0b0e984cc8f599646289a55d66ca0b89be355a) | [after](https://github.com/elastic/kibana/commit/e01f85d0de69f6a5f05dde1a51b3312e76a25cca) | diff |
| --- | --- | --- | --- |
| `lens` | 181 | 185 | +4 |


</details>

### History
* :green_heart: [Build #138430](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/) succeeded 6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273
* :broken_heart: [Build #138408](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/) failed fd994e6a382edc722001e3657ed7593c703986d2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/","number":"138408","commit":"fd994e6a382edc722001e3657ed7593c703986d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/","number":"138430","commit":"6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/","number":"138666","commit":"e01f85d0de69f6a5f05dde1a51b3312e76a25cca"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

stratoula commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/)
* Commit: ddc40da0967c646cc022773ded7b52f79a1867ab
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105571/ddc40da0967c646cc022773ded7b52f79a1867ab)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/7dcb136c688eb436b63612596daefeb8d19aa5bc) | [after](https://github.com/elastic/kibana/commit/ddc40da0967c646cc022773ded7b52f79a1867ab) | diff |
| --- | --- | --- | --- |
| `lens` | 703 | 704 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/7dcb136c688eb436b63612596daefeb8d19aa5bc) | [after](https://github.com/elastic/kibana/commit/ddc40da0967c646cc022773ded7b52f79a1867ab) | diff |
| --- | --- | --- | --- |
| `lens` | 1.5MB | 1.5MB | +9.3KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/7dcb136c688eb436b63612596daefeb8d19aa5bc) | [after](https://github.com/elastic/kibana/commit/ddc40da0967c646cc022773ded7b52f79a1867ab) | diff |
| --- | --- | --- | --- |
| `lens` | 181 | 185 | +4 |


</details>

### History
* :green_heart: [Build #138666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/) succeeded e01f85d0de69f6a5f05dde1a51b3312e76a25cca
* :green_heart: [Build #138430](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/) succeeded 6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273
* :broken_heart: [Build #138408](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/) failed fd994e6a382edc722001e3657ed7593c703986d2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/","number":"138408","commit":"fd994e6a382edc722001e3657ed7593c703986d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/","number":"138430","commit":"6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/","number":"138666","commit":"e01f85d0de69f6a5f05dde1a51b3312e76a25cca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/","number":"138719","commit":"ddc40da0967c646cc022773ded7b52f79a1867ab"}]}
PIPELINE-->

dej611 reviewed (commented):
I'm not convinced by the *Alignment* field: 

<img width="359" alt="Screenshot 2021-07-15 at 16 33 17" src="https://user-images.githubusercontent.com/924948/125806581-557844a1-e5db-48b0-a88b-176b2f4fc374.png">

would it be better to have a group of 4 buttons pointing to each screen corner? (quick mockup):
<img width="359" alt="Screenshot 2021-07-15 at 16 33 17" src="https://user-images.githubusercontent.com/924948/125808909-9a0f0a9d-f86d-4f49-a099-eeea1dbc5953.png">

Or even something more minimal with just the arrow pointing to the corner.

Is the *Show value* feature broken? I've tried to enable it but it does not work.

I find the *Number of columns* option very useful: it would be nice to extend it also to the external legend positioning (maybe followup)

ghudgins commented:
@MichaelMarcialis what do you think of @dej611's question here https://github.com/elastic/kibana/pull/105571#pullrequestreview-707268422

wylieconlon commented on x-pack/plugins/lens/public/shared_components/legend_location_settings.tsx:
```diff
@@ -0,0 +1,223 @@
+    <EuiRange
+      data-test-subj="lens-legend-location-columns-slider"
+      value={inputValue}
+      min={0}
```
Shouldn't the min be `1` here?

stratoula replied:
You are right!

wylieconlon reviewed (approved):
I agree with @dej611 that the current horizontal/vertical controls are using confusing icons. EUI provides 4 icons like `editorPositionTopLeft` which we should probably use to control _both_ horizontal and vertical position at the same time.

@dej611 I tested the "show value" option and it works correctly.

Is it an elastic-charts feature that there is a background color assigned on hover? Is this something that works in dark mode?

<img width="223" alt="Screen Shot 2021-07-15 at 2 28 35 PM" src="https://user-images.githubusercontent.com/666475/125839712-a2f55534-4b56-4f72-bf2e-6067dff39aad.png">

dej611 commented:
> Is it an elastic-charts feature that there is a background color assigned on hover? Is this something that works in dark mode?
> 
> <img alt="Screen Shot 2021-07-15 at 2 28 35 PM" width="223" src="https://user-images.githubusercontent.com/666475/125839712-a2f55534-4b56-4f72-bf2e-6067dff39aad.png">

I've forgotten to report one point about that: there's some blending issue with the background color and the chart color.

<img width="676" alt="Screenshot 2021-07-15 at 16 36 03" src="https://user-images.githubusercontent.com/924948/125950621-239b7975-b2e9-4a91-98ec-bf575cb24a62.png">

MichaelMarcialis commented:
> @MichaelMarcialis what do you think of @dej611's question here [#105571 (review)](https://github.com/elastic/kibana/pull/105571#pullrequestreview-707268422)

I agree with @dej611's suggested change. As @wylieconlon already noted, we can use the following EUI icons to represent each corner in a single `EuiButtonGroup`:

- `editorPositionBottomLeft`
- `editorPositionBottomRight`
- `editorPositionTopLeft`
- `editorPositionTopRight`

MichaelMarcialis reviewed (commented):
This is looking good, @stratoula! I've left a few comments for your review:

- Rather than using two separate/distinct form rows and labels for "Position" and "Alignment," could we just have a single form row called "Alignment," and change the contained button group depending on which location modes (inside/outside) is selected? I think doing so will be less confusing to users, as the two perform very similar functions, despite having different button group options.

- Currently, the "Number of columns" form row disappears when the "Location" is set to "Outside." However, other form inputs within this menu become disabled rather than hidden when they are unavailable (but can still be re-enabled in the current context). As such, can we instead disable the "Number of columns" form input when "Location" is set to "Outside," rather than hiding it altogether?

- For situations when a form input is disabled due to currently selected options, can we indicate to users how they can go about re-enabling that form input? For example, when "Display" is set to "Hide," could we have tooltips that appear on hover of the "Location," "Alignment," "Number of columns," and "Show value" inputs that states `Requires legend to be shown`? Similarly, when "Location" is set to "Outside", the now disabled "Number of columns" input can have a tooltip on hover that states `Requires legend to be located inside visualization`.

stratoula commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139255/)
* Commit: 51f8fcce05782baca0cee5e1fad5bb05a1dd87e0
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105571/51f8fcce05782baca0cee5e1fad5bb05a1dd87e0)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/e9988b1cc2c8f322f63f441891c9032c9b9f25ce) | [after](https://github.com/elastic/kibana/commit/51f8fcce05782baca0cee5e1fad5bb05a1dd87e0) | diff |
| --- | --- | --- | --- |
| `lens` | 703 | 704 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e9988b1cc2c8f322f63f441891c9032c9b9f25ce) | [after](https://github.com/elastic/kibana/commit/51f8fcce05782baca0cee5e1fad5bb05a1dd87e0) | diff |
| --- | --- | --- | --- |
| `lens` | 1.5MB | 1.5MB | +9.3KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/e9988b1cc2c8f322f63f441891c9032c9b9f25ce) | [after](https://github.com/elastic/kibana/commit/51f8fcce05782baca0cee5e1fad5bb05a1dd87e0) | diff |
| --- | --- | --- | --- |
| `lens` | 181 | 185 | +4 |


</details>

### History
* :green_heart: [Build #138719](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/) succeeded ddc40da0967c646cc022773ded7b52f79a1867ab
* :green_heart: [Build #138666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/) succeeded e01f85d0de69f6a5f05dde1a51b3312e76a25cca
* :green_heart: [Build #138430](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/) succeeded 6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273
* :broken_heart: [Build #138408](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/) failed fd994e6a382edc722001e3657ed7593c703986d2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/","number":"138408","commit":"fd994e6a382edc722001e3657ed7593c703986d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/","number":"138430","commit":"6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/","number":"138666","commit":"e01f85d0de69f6a5f05dde1a51b3312e76a25cca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/","number":"138719","commit":"ddc40da0967c646cc022773ded7b52f79a1867ab"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139255/","number":"139255","commit":"51f8fcce05782baca0cee5e1fad5bb05a1dd87e0"}]}
PIPELINE-->

stratoula commented on x-pack/plugins/lens/public/shared_components/legend_location_settings.tsx:
```diff
@@ -0,0 +1,223 @@
+  setValue: (value: number) => void;
+  isDisabled: boolean;
+}) => {
+  const { inputValue, handleInputChange } = useDebouncedValue({ value, onChange: setValue });
```
@dej611 I will follow Wylie's suggestion and change the min to 1

stratoula commented:
> Is it an elastic-charts feature that there is a background color assigned on hover? Is this something that works in dark mode?
> 
> <img alt="Screen Shot 2021-07-15 at 2 28 35 PM" width="223" src="https://user-images.githubusercontent.com/666475/125839712-a2f55534-4b56-4f72-bf2e-6067dff39aad.png">

Yes @wylieconlon it is.
Here how it works in dark mode
![image](https://user-images.githubusercontent.com/17003240/126139929-5e87e404-d7f6-4394-9892-6bb4d089b983.png)

stratoula commented on x-pack/plugins/lens/public/shared_components/legend_location_settings.tsx:
```diff
@@ -0,0 +1,223 @@
+            fullWidth
+            display="rowCompressed"
+          >
+            <FloatingColumnsSlider
+              value={floatingColumns ?? DEFAULT_FLOATING_COLUMNS}
+              setValue={onFloatingColumnsChange}
+              isDisabled={isDisabled}
+            />
```
I will change it to EuiFieldNumber!

stratoula commented:
Thank you all for the alignment buttons suggestion. It is much better that way!
![image](https://user-images.githubusercontent.com/17003240/126158563-f67178f9-f827-43f0-b278-5aec9987f735.png)

@MichaelMarcialis I have addressed all your comments (I hope). 
- Changed the alignment icon buttons to the suggested ones
- Position is renamed and used Alignement everywhere
- Added tooltips to inform the users on how to enable these settings
- Changed the position icons order to what you suggested
- Changed the slider to number input

Can you take another look? Thanx :)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139340/)
* Commit: 0693aff998cd26e024a9ecdae3ddf40d221f76fa
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105571/0693aff998cd26e024a9ecdae3ddf40d221f76fa)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/e9988b1cc2c8f322f63f441891c9032c9b9f25ce) | [after](https://github.com/elastic/kibana/commit/0693aff998cd26e024a9ecdae3ddf40d221f76fa) | diff |
| --- | --- | --- | --- |
| `lens` | 703 | 704 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e9988b1cc2c8f322f63f441891c9032c9b9f25ce) | [after](https://github.com/elastic/kibana/commit/0693aff998cd26e024a9ecdae3ddf40d221f76fa) | diff |
| --- | --- | --- | --- |
| `lens` | 1.5MB | 1.5MB | +10.8KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/e9988b1cc2c8f322f63f441891c9032c9b9f25ce) | [after](https://github.com/elastic/kibana/commit/0693aff998cd26e024a9ecdae3ddf40d221f76fa) | diff |
| --- | --- | --- | --- |
| `lens` | 181 | 185 | +4 |


</details>

### History
* :green_heart: [Build #139255](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139255/) succeeded 51f8fcce05782baca0cee5e1fad5bb05a1dd87e0
* :green_heart: [Build #138719](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/) succeeded ddc40da0967c646cc022773ded7b52f79a1867ab
* :green_heart: [Build #138666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/) succeeded e01f85d0de69f6a5f05dde1a51b3312e76a25cca
* :green_heart: [Build #138430](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/) succeeded 6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273
* :broken_heart: [Build #138408](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/) failed fd994e6a382edc722001e3657ed7593c703986d2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138408/","number":"138408","commit":"fd994e6a382edc722001e3657ed7593c703986d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/","number":"138430","commit":"6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/","number":"138666","commit":"e01f85d0de69f6a5f05dde1a51b3312e76a25cca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/","number":"138719","commit":"ddc40da0967c646cc022773ded7b52f79a1867ab"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139255/","number":"139255","commit":"51f8fcce05782baca0cee5e1fad5bb05a1dd87e0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139340/","number":"139340","commit":"0693aff998cd26e024a9ecdae3ddf40d221f76fa"}]}
PIPELINE-->

wylieconlon reviewed (commented):
One more comment, maybe for @markov00: it seems like the inner legend + columns option leads to overflow pretty often, especially when the # of columns is higher than the available width. Is this something that the charts library can detect and solve automatically?

markov00 commented:
> One more comment, maybe for @markov00: it seems like the inner legend + columns option leads to overflow pretty often, especially when the # of columns is higher than the available width. Is this something that the charts library can detect and solve automatically?

Something like that can be done probably, but my suggestion is to reduce the max number of columns to 4 max 5 for now.

stratoula commented:
> Something like that can be done probably, but my suggestion is to reduce the max number of columns to 4 max 5 for now.
I agree, I reduced it to max5 for now

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139653/)
* Commit: 4b2819ca727f10de34e4f275ba69d42ac4f32deb
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105571/4b2819ca727f10de34e4f275ba69d42ac4f32deb)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/1e30148f59c84ca8f71a07481deee9d84911c9b3) | [after](https://github.com/elastic/kibana/commit/4b2819ca727f10de34e4f275ba69d42ac4f32deb) | diff |
| --- | --- | --- | --- |
| `lens` | 703 | 704 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/1e30148f59c84ca8f71a07481deee9d84911c9b3) | [after](https://github.com/elastic/kibana/commit/4b2819ca727f10de34e4f275ba69d42ac4f32deb) | diff |
| --- | --- | --- | --- |
| `lens` | 1.5MB | 1.5MB | +10.8KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/1e30148f59c84ca8f71a07481deee9d84911c9b3) | [after](https://github.com/elastic/kibana/commit/4b2819ca727f10de34e4f275ba69d42ac4f32deb) | diff |
| --- | --- | --- | --- |
| `lens` | 181 | 185 | +4 |


</details>

### History
* :green_heart: [Build #139340](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139340/) succeeded 0693aff998cd26e024a9ecdae3ddf40d221f76fa
* :green_heart: [Build #139255](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139255/) succeeded 51f8fcce05782baca0cee5e1fad5bb05a1dd87e0
* :green_heart: [Build #138719](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/) succeeded ddc40da0967c646cc022773ded7b52f79a1867ab
* :green_heart: [Build #138666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/) succeeded e01f85d0de69f6a5f05dde1a51b3312e76a25cca
* :green_heart: [Build #138430](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/) succeeded 6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138430/","number":"138430","commit":"6e43e68cdf3b7a0a85e38f1dcf1fcbf3f1277273"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/","number":"138666","commit":"e01f85d0de69f6a5f05dde1a51b3312e76a25cca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/","number":"138719","commit":"ddc40da0967c646cc022773ded7b52f79a1867ab"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139255/","number":"139255","commit":"51f8fcce05782baca0cee5e1fad5bb05a1dd87e0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139340/","number":"139340","commit":"0693aff998cd26e024a9ecdae3ddf40d221f76fa"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139653/","number":"139653","commit":"4b2819ca727f10de34e4f275ba69d42ac4f32deb"}]}
PIPELINE-->

MichaelMarcialis reviewed (approved):
Thanks for making those requested changes, @stratoula. The addition of the tooltips did create one small bug, where when the tooltips are present the elements contained within collapse down (due to the tooltip container being inline by default). You should be able to fix this by applying a `display="block"` prop to the related `EuiToolTip` components.

![legend](https://user-images.githubusercontent.com/3884767/126338927-c4a396c5-8b88-43a6-b496-833869c27eed.gif)

Otherwise, this looks good to me once the above is addressed. Approving now so I don't hold you up.

dej611 commented:
> > Something like that can be done probably, but my suggestion is to reduce the max number of columns to 4 max 5 for now.
> 
> I agree, I reduced it to max5 for now

Just tested and the 5 max limit imposed is only checked if the user uses the input number steppers handles. Probably a validation should be set on top of it for user's manual input?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139760/)
* Commit: 5933ab762b7f8e37310d634811a24b4afe6cf3c5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105571/5933ab762b7f8e37310d634811a24b4afe6cf3c5)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/1e30148f59c84ca8f71a07481deee9d84911c9b3) | [after](https://github.com/elastic/kibana/commit/5933ab762b7f8e37310d634811a24b4afe6cf3c5) | diff |
| --- | --- | --- | --- |
| `lens` | 703 | 704 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/1e30148f59c84ca8f71a07481deee9d84911c9b3) | [after](https://github.com/elastic/kibana/commit/5933ab762b7f8e37310d634811a24b4afe6cf3c5) | diff |
| --- | --- | --- | --- |
| `lens` | 1.5MB | 1.5MB | +10.9KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/1e30148f59c84ca8f71a07481deee9d84911c9b3) | [after](https://github.com/elastic/kibana/commit/5933ab762b7f8e37310d634811a24b4afe6cf3c5) | diff |
| --- | --- | --- | --- |
| `lens` | 181 | 185 | +4 |


</details>

### History
* :green_heart: [Build #139653](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139653/) succeeded 4b2819ca727f10de34e4f275ba69d42ac4f32deb
* :green_heart: [Build #139340](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139340/) succeeded 0693aff998cd26e024a9ecdae3ddf40d221f76fa
* :green_heart: [Build #139255](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139255/) succeeded 51f8fcce05782baca0cee5e1fad5bb05a1dd87e0
* :green_heart: [Build #138719](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/) succeeded ddc40da0967c646cc022773ded7b52f79a1867ab
* :green_heart: [Build #138666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/) succeeded e01f85d0de69f6a5f05dde1a51b3312e76a25cca

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138666/","number":"138666","commit":"e01f85d0de69f6a5f05dde1a51b3312e76a25cca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138719/","number":"138719","commit":"ddc40da0967c646cc022773ded7b52f79a1867ab"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139255/","number":"139255","commit":"51f8fcce05782baca0cee5e1fad5bb05a1dd87e0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139340/","number":"139340","commit":"0693aff998cd26e024a9ecdae3ddf40d221f76fa"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139653/","number":"139653","commit":"4b2819ca727f10de34e4f275ba69d42ac4f32deb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139760/","number":"139760","commit":"5933ab762b7f8e37310d634811a24b4afe6cf3c5"}]}
PIPELINE-->
